skxray/core/xsvs.py

```diff
@@ -0,0 +1,289 @@
+    bin_edges = np.zeros((num_times, num_roi), dtype=object)
+    for i in range(num_times):
+        for j in range(num_roi):
+            bin_edges[i, j] = np.arange(max_cts*2**i)
+
+    start_time = time.time()  # used to log the computation time (optionally)
+
+    for i, images in enumerate(image_sets):
+        # Ring buffer, a buffer with periodic boundary conditions.
+        # Images must be keep for up to maximum delay in buf.
+        buf = np.zeros([num_times, timebin_num],
+                       dtype=np.object)  # matrix of buffers
+
+        # to track processing each time level
+        track_level = np.zeros(num_times)
+
+        # to increment buffer
+        cur = np.ones(num_times)*timebin_num
+
+        # to track how many images processed in each level
+        img_per_level = np.zeros(num_times, dtype=np.int64)
+
+        prob_k = np.zeros([num_times, num_roi], dtype=np.object)
+        prob_k_pow = np.zeros([num_times, num_roi], dtype=np.object)
+
+        for n, img in enumerate(images):
+            cur[0] = (1 + cur[0]) % timebin_num
+            # read each frame
+            # Put the image into the ring buffer.
+            buf[0, cur[0] - 1] = (np.ravel(img))[indices]
+
+            _process(num_roi, 0, cur[0] - 1, buf, img_per_level, labels,
+                     max_cts, bin_edges[0, 0], prob_k, prob_k_pow)
+
+            # check whether the number of levels is one, otherwise
+            # continue processing the next level
+            processing = num_times > 1
+            level = 1
+
+            while processing:
+                if not track_level[level]:
+                    track_level[level] = 1
+                    processing = 0
+                else:
+                    prev = 1 + (cur[level - 1] - 2) % timebin_num
+                    cur[level] = 1 + cur[level] % timebin_num
+
+                    buf[level, cur[level]-1] = (buf[level-1,
+                                                    prev-1] +
+                                                buf[level-1,
+                                                    cur[level - 1] - 1])
+                    track_level[level] = 0
+
+                    _process(num_roi, level, cur[level]-1, buf, img_per_level,
+                             labels, max_cts, bin_edges[level, 0], prob_k,
+                             prob_k_pow)
+                    level += 1
+                    # Checking whether there is next level for processing
+                    processing = level < num_times
+
+            prob_k_all += (prob_k - prob_k_all)/(i + 1)
+            prob_k_pow_all += (prob_k_pow - prob_k_pow_all)/(i + 1)
+
+            prob_k_std_dev = np.power((prob_k_pow_all -
+                                       np.power(prob_k_all, 2)), .5)
+
+    # ending time for the process
+    end_time = time.time()
+
+    logger.info("Processing time for XSVS took {0} seconds."
+                "".format(end_time - start_time))
```

Comment:
Can you rewrite this as 

```
    logger.info("Processing time for XSVS took %s seconds."
                "", (time.time() - start_time))
```

and remove the `end_time = time.time()` line above?

If you write logging messages with the `%` syntax, they are only evaluated if the logger's level is set such that, in this case, `info`, messages are logged